ledger/eval/eval_test.go

```diff
@@ -291,7 +291,8 @@ func TestTransactionGroupWithTracer(t *testing.T) {
 			blkHeader, err := l.BlockHdr(basics.Round(0))
 			require.NoError(t, err)
 			newBlock := bookkeeping.MakeBlock(blkHeader)
-			eval, err := l.StartEvaluator(newBlock.BlockHeader, 0, 0)
+			tracer := &mocktracer.Tracer{}
+			eval, err := l.StartEvaluator(newBlock.BlockHeader, 0, 0, tracer)
```

Comment:
Is this test being modified purely because there is now a prefered way to attach a tracer? I'm trying to understand what new behavior is being tested.

Reply:
It's both the new way of attaching the tracer, but also the new Before/After Block hooks. Here I've modified how the tracer is attached, and below you can see I've added the `mocktracer.AfterBlock(eval.block.Round()),` event to assert that the tracer was called as expected.